sorting/dnf_sort.c

```diff
@@ -0,0 +1,121 @@
+/**
+ * @file
+ * @brief Program to perform [dutch national
+ * flag](https://en.wikipedia.org/wiki/Dutch_national_flag_problem) of a target
+ * value in a given {0, 1, 2} containing array.
+ * @authors [Bijan Riesenberg](https://github.com/Data-Hero) Iterative
+ */
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+/**
+ * @brief Dutch National Flag 3-Way inplace sort algorithm
+ * @param arr array containting elements from {0, 1, 2} to be sorted
+ * @param size size of the array
+ */
+void dnfSort(int8_t *arr, int size)
+{
+    int low = 0, mid = 0, high = size - 1;
+    int8_t temp = 0;
+    while (mid <= high)
+    {
+        switch (arr[mid])
+        {
+        case 0:
+            temp = arr[low];
+            arr[low++] = arr[mid];
+            arr[mid++] = temp;
+            break;
+        case 1:
+            ++mid;
+            break;
+        case 2:
+            temp = arr[mid];
+            arr[mid] = arr[high];
+            arr[high--] = temp;
+            break;
+        default:
+            break;
+        }
+    }
+}
+
+/**
+ * @brief Print an array containing int8_t elements
+ *
+ * @param arr array containting elements from {0, 1, 2} (int8_t) to be printed
+ * @param size size of the array
+ */
+void printArray(int8_t *arr, int size)
+{
+    for (int c = 0; c < size; c++)
+    {
+        c != size - 1 ? printf("%" PRIi8 ", ", arr[c])
+                      : printf("%" PRIi8 "\n", arr[c]);
+    }
+}
+
+/**
+ * @brief Self-test implementations
+ * @returns void
+ */
+static void test()
+{
+    int size = 6;
+    int8_t *arr1 = (int8_t *)malloc(sizeof(int8_t) * size);
```

Comment:
In the self-test implementations, we use `assert` to ensure the program works properly. Do you think you could use that here in this function? You require the `assert.h` library. Let us know if you need any help. :)